stan/math/prim/err/check_matching_dims.hpp

```diff
@@ -70,6 +70,43 @@ inline void check_matching_dims(const char* function, const char* name1,
   check_matching_dims(function, name1, y1, name2, y2);
 }
 
+/**
+ * Check if the two scalars are of the same size. The check always succeeds.
+ * @tparam T1 type of the first scalar
+ * @tparam T2 type of the second scalar
+ * @param function name of function (for error messages)
+ * @param name1 variable name for the first scalar (for error messages)
+ * @param y1 first scalar to test
+ * @param name2 variable name for the second scalar (for error messages)
+ * @param y2 second scalar to test
+ */
+template <typename T1, typename T2,
+          require_all_stan_scalar_t<T1, T2>* = nullptr>
+inline void check_matching_dims(const char* function, const char* name1,
+                                const T1& y1, const char* name2, const T2& y2) {
+}
+
+/**
+ * Check if the two containers have the same dimensions.
+ * @tparam Mat1 type of the first container
+ * @tparam Mat2 type of the second container
+ * @param function name of function (for error messages)
+ * @param name1 variable name for the first container (for error messages)
+ * @param y1 first container to test
+ * @param name2 variable name for the second container (for error messages)
+ * @param y2 second container to test
+ * @throw <code>std::invalid_argument</code> if the dimensions of the
+ *    containers do not match
+ */
+template <typename T1, typename T2>
+inline void check_matching_dims(const char* function, const char* name1,
+                                const std::vector<T1>& y1, const char* name2,
+                                const std::vector<T2>& y2) {
+  check_size_match(function, name1, y1.size(), name2, y2.size());
```

Comment:
You could simplify this by using `dims()`: https://github.com/stan-dev/math/blob/develop/stan/math/prim/fun/dims.hpp

No need for the scalar overload then, just call dims and iterate over the resulting vector.

I know there is not much code here, so not a huge deal, but still makes sense not to duplicate.

Reply:
You're right. I wonder why I didn't do that... I think I wanted to preserve `rows of`/`columns of` in the error message when the base type is a matrix. But the error messages have are half-baked anyway...

Reply:
I think with calling dims the error message might actually be better as you could mention on which dimension sizes differ.